### ISSUE TITLE
Issue with PSR-12 behavior not according specification with spacing for anonymous classes

### ISSUE THREAD

**Issue by rikvdh** [PHPCSStandards/PHP_CodeSniffer#569]
## Describe the bug

Found while developing tests for PER2.0, I got the error: `PSR12.Classes.AnonClassDeclaration.SpaceAfterKeyword`, but when diving into specifications for PSR-12, I could not found anything related to spacing being required there.

### Code sample
```php
<?php

$foo = new class() {};
//             ^^
```

Here an error is raised about needing a space between the arguments for the class and the first opening-parenthesis. But nowhere in the specification, nor the errata, I can find on required spacing after class keywords.

What it should be is a bit ambiguous, but when it is not specified it should not be checked. It is 'class instantiation' and should require parentheses. But the space there should not be checked imo.

### Custom ruleset
N/A

## Expected behavior

I think we need to mute this error.

## Please confirm

- [X] I have searched the issue list and am not opening a duplicate issue.
- [X] I have read the [Contribution Guidelines](https://github.com/PHPCSStandards/PHP_CodeSniffer/blob/master/.github/CONTRIBUTING.md) and this is not a [support question](https://github.com/PHPCSStandards/PHP_CodeSniffer/discussions).
- [X] I confirm that this bug is a bug in PHP_CodeSniffer and not in one of the external standards.
- [X] I have verified the issue still exists in the `master` branch of PHP_CodeSniffer.

**Comment by jrfnl** [comment]
This was previously discussed in https://github.com/squizlabs/PHP_CodeSniffer/issues/3200 and I don't think anything has changed in PSR12 since that discussion.

**Comment by rikvdh** [comment]
There is now in PER 2.0 a clear example which is then 'wrong'. I've submitted a proposal to fix this in the spec.

https://github.com/php-fig/per-coding-style/pull/90

**Comment by jrfnl** [comment]
Also related: https://github.com/PHPCSStandards/PHP_CodeSniffer/pull/47

**Comment by jrfnl** [comment]
@rikvdh As the PR to PER has been merged now, do you agree we can close this issue ?

**Comment by rikvdh** [comment]
Yes, could be closed now.
